ntoskrnl/ke/i386/thrdini.c

```diff
@@ -497,6 +497,9 @@ KiDispatchInterrupt(VOID)
         NewThread->State = Running;
         OldThread->WaitReason = WrDispatchInt;
 
+        /* Acquire PRCB lock */
```

Comment:
It would be of interest to follow what the x64 code does:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ke/amd64/traphandler.c;hb=9c1cfe058706ff31dbcd3c7b53eec879f66c8030#l58
and put that lock-acquire above.

Also it may be worth fixing (even if it's unused) this code here, for consistency:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ke/arm/thrdini.c;hb=9c1cfe058706ff31dbcd3c7b53eec879f66c8030#l325
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ke/arm/trapc.c;hb=9c1cfe058706ff31dbcd3c7b53eec879f66c8030#l321

Reply:
OK @HBelusca. Will test it as well. :slightly_smiling_face: 
Waiting for the amd64 artifacts in my test branch:
https://github.com/oleg-dubinskiy/reactos/tree/amd64_prcb_lock_test

Reply:
Ah, lol @HBelusca! The lock is already added above in this function on x64! :smile: 
https://github.com/reactos/reactos/blob/master/ntoskrnl/ke/amd64/traphandler.c#L60
See the following commit also:
https://github.com/reactos/reactos/commit/a011d19ed2ae2345fbfcd429590e1bec9194fa6e
So it seems to be correct. :smirk:

Reply:
> The lock is already added above in this function on x64!

Hermès' suggestion is about `arm`, not `amd64`...

Reply:
> > The lock is already added above in this function on x64!
> 
> Hermès' suggestion is about `arm`, not `amd64`...

Okay, added the fix for arm too @SergeGautherie. :slightly_smiling_face:

Reply:
Looks that the following code is commented out:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ke/arm/trapc.c;hb=9c1cfe058706ff31dbcd3c7b53eec879f66c8030#l321
but it _may_ also be of interest to add the locking there too, for consistency (in the hypothetical case the code gets revived).